sdk/ai/azure-ai-projects/samples/agents/sample_agents_morningstar.py

```diff
@@ -0,0 +1,168 @@
+# pylint: disable=line-too-long,useless-suppression
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+DESCRIPTION:
+    This sample demonstrates how to use agent operations with the
+    Morningstar OpenAPI tool, from the Azure Agents service using a synchronous client.
+    To learn more about OpenAPI specs, visit https://learn.microsoft.com/openapi
+    Link to the Morningstar Agent OpenAPI spec: https://developer.morningstar.com/direct-web-services/documentation/intelligence-engine/apps/morningstar-agent-api
+USAGE:
+    python sample_agents_openapi_morningstar.py
+    Before running the sample:
+    pip install azure-ai-projects azure-identity jsonref
+    Set these environment variables with your own values:
+    1) PROJECT_CONNECTION_STRING - The project connection string, as found in the overview page of your
+       Azure AI Foundry project.
+    2) MODEL_DEPLOYMENT_NAME - The deployment name of the AI model, as found under the "Name" column in
+       the "Models + endpoints" tab in your Azure AI Foundry project.
+    3) PROJECT_CONNECTION_ID - The project connection id of your AI Foundry project.
+    You'll also need a JWT token to authorize with the Morningstar service.
+    You can fetch this JWT token by using the below code:
+    import requests
+    from requests.auth import HTTPBasicAuth
+    url = "https://www.us-api.morningstar.com/token/oauth"
+    auth = HTTPBasicAuth("john.doe@example.com", "SecureP@ssw0rd")
+    response = requests.post(url, auth=auth)
+    print(response.text)
+    Note: This token expires, please remember to refresh your credentials.
+"""
+
+import os
+import jsonref
+from azure.ai.projects import AIProjectClient
+from azure.ai.projects.models import (
+    OpenApiConnectionAuthDetails,
+    OpenApiConnectionSecurityScheme,
+    OpenApiTool,
+)
+from azure.identity import DefaultAzureCredential
+
+project_client = AIProjectClient.from_connection_string(
+    credential=DefaultAzureCredential(),
+    conn_str=os.environ["PROJECT_CONNECTION_STRING"],
+)
+
+with open("./iep_openapi.json", "r") as f:
```

Comment:
`iep_openapi.json` if missing from this PR

Reply:
@dargilco it is available here: https://github.com/Azure/azure-sdk-for-python/pull/40389/files#diff-86f2d9ad96c80f44372391648ceddb1fe7b0176e2c425d015757fb9f4ea964ecR12

Reply:
I added an additional comment to help users, let me know if you'd like to make more changes.